src/app/features/event/event.state.service.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ENDPOINTS } from '@shared/constants';
+
+import { BehaviorSubject } from 'rxjs';
+import { Maybe } from '@shared/utility-types';
+import { EventModule } from './event.module';
+import { EventCardDTO, EventDetailsDTO } from '.';
+
+export type EventState = {
+  eventDetails: Maybe<EventDetailsDTO>;
+  eventsList: EventCardDTO[];
+};
+
+const eventStateDefault = {
+  eventDetails: null,
+  eventsList: [],
+};
+
+@Injectable({
+  providedIn: EventModule,
+})
+export class EventStateService {
+  private eventState$$ = new BehaviorSubject<EventState>(eventStateDefault);
+  constructor(private http: HttpClient) {}
```

Comment:
Moge się mylic dlatego tylko pytanie z mojej strony, czy nie powinienes użyć Inject() do wstrzykniecie http, wiem ze działa tak samo jak w użyciu w constructorze ale kojarzę z wykładów, że powinniśmy się przyzwyczajać właśnie do zapisu bez uzycia constructora

Reply:
Sugestia dobra, tylko odnosi się do taska z event serwisem także u siebie zostawiam tak jak jest, jako że nie jest to błąd.